backends/cuda-gen/ceed-cuda-gen-operator-build.cpp

```diff
@@ -1130,16 +1140,14 @@ extern "C" int CeedCudaGenOperatorBuild(CeedOperator op) {
     CeedChkBackend(ierr);
     if (emode==CEED_EVAL_GRAD)
     {
+      code << "    CeedScalar r_tt"<<i<<"[ncomp_out_"<<i<<"*Q1d];\n";
       if (useCollograd) {
         //Accumulator for gradient slices
-        code << "    CeedScalar r_tt"<<i<<"[ncomp_out_"<<i<<"*Q1d];\n";
         code << "    for (CeedInt i = 0; i < ncomp_out_"<<i<<"; ++i) {\n";
         code << "      for (CeedInt j = 0; j < Q1d; ++j) {\n";
         code << "        r_tt"<<i<<"[j + i*Q1d] = 0.0;\n";
         code << "      }\n";
         code << "    }\n";
-      } else {
-        code << "    CeedScalar r_tt"<<i<<"[ncomp_out_"<<i<<"*Dim*Q1d];\n";
```

Comment:
It looks like these changes have not yet made it into `hip-gen` -- could this be the reason for the failing fluids test on CI?

Reply:
I don't think this should create issues, the size of the array is just bigger than needed if I'm not mistaken. I'll take a closer look at the Hip code.

Reply:
I propagated the change in [9781974](https://github.com/CEED/libCEED/pull/1021/commits/97819745621ba5e592629ee16c7baa04faf77f86).